tasks/dashboards.yml

```diff
@@ -13,12 +13,22 @@
     # Use curl to solve issue #77
     - name: download grafana dashboard from grafana.net to local directory
       command: >
-        curl --fail --compressed
-        https://grafana.com/api/dashboards/{{ item.dashboard_id }}/revisions/{{ item.revision_id }}/download
-        -o {{ _tmp_dashboards.path }}/{{ item.dashboard_id }}.json
+        curl --fail --compressed {{ url }} -o {{ _tmp_dashboards.path }}/{{ filename }}
       args:
-        creates: "{{ _tmp_dashboards.path }}/{{ item.dashboard_id }}.json"
+        creates: "{{ _tmp_dashboards.path }}/{{ filename }}"
         warn: false
+      vars:
+        source: "{{ 'url' if 'url' in item else 'grafana' }}"
+        filename: >-
+          {%- if source == "url" -%}
+          {{ item.url | urlsplit('path') | basename }}
+          {%- else -%}
+          {{ item.dashboard_id }}.json
+          {%- endif -%}
```

Comment:
This logic is used twice, would it be possible to do it once in preflight.yml and reuse its output?